homeassistant/components/guardian/switch.py

```diff
@@ -23,52 +24,63 @@
 
 SWITCH_KIND_VALVE = "valve"
 
-SWITCH_DESCRIPTION_VALVE = SwitchEntityDescription(
-    key=SWITCH_KIND_VALVE,
-    name="Valve controller",
-    icon="mdi:water",
+
+@dataclass
+class ValveControllerSwitchDescription(
+    SwitchEntityDescription, ValveControllerEntityDescription
+):
+    """Describe a Guardian valve controller switch."""
+
+
+VALVE_CONTROLLER_DESCRIPTIONS = (
+    ValveControllerSwitchDescription(
+        key=SWITCH_KIND_VALVE,
+        name="Valve controller",
+        icon="mdi:water",
+        api_category=API_VALVE_STATUS,
+    ),
 )
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up Guardian switches based on a config entry."""
+    client = hass.data[DOMAIN][entry.entry_id][DATA_CLIENT]
+    valve_controller_coordinators = hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR]
+
     async_add_entities(
         [
             ValveControllerSwitch(
-                entry,
-                hass.data[DOMAIN][entry.entry_id][DATA_CLIENT],
-                hass.data[DOMAIN][entry.entry_id][DATA_COORDINATOR],
+                entry, valve_controller_coordinators, description, client
             )
+            for description in VALVE_CONTROLLER_DESCRIPTIONS
         ]
     )
 
 
 class ValveControllerSwitch(ValveControllerEntity, SwitchEntity):
     """Define a switch to open/close the Guardian valve."""
 
+    entity_description: ValveControllerSwitchDescription
+
     def __init__(
         self,
         entry: ConfigEntry,
+        coordinators: dict[str, GuardianDataUpdateCoordinator],
+        description: ValveControllerSwitchDescription,
         client: Client,
-        coordinators: dict[str, DataUpdateCoordinator],
     ) -> None:
         """Initialize."""
-        super().__init__(entry, coordinators, SWITCH_DESCRIPTION_VALVE)
+        super().__init__(entry, coordinators, description)
 
         self._attr_is_on = True
         self._client = client
 
-    async def _async_continue_entity_setup(self) -> None:
-        """Register API interest (and related tasks) when the entity is added."""
-        self.async_add_coordinator_update_listener(API_VALVE_STATUS)
-
     @callback
     def _async_update_from_latest_data(self) -> None:
         """Update the entity."""
-        self._attr_available = self.coordinators[API_VALVE_STATUS].last_update_success
```

Comment:
```python
        data = self.coordinator.data
        self._attr_is_on = data["state"] in ON_STATES
        self._attr_extra_state_attributes.update(
            {
                ATTR_AVG_CURRENT: data["average_current"],
                ATTR_INST_CURRENT: data["instantaneous_current"],
                ATTR_INST_CURRENT_DDT: data["instantaneous_current_ddt"],
                ATTR_TRAVEL_COUNT: data["travel_count"],
            }
        )
```

Reply:
defined as as set to avoid a linear search

```python

ON_STATES = {
    "start_opening",
    "opening",
    "finish_opening",
    "opened",
}
```